src/components/Menu.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+import ".././styles/styles.scss";
+
+class Menu extends React.Component<any, any> {
+    constructor(props) {
+        super(props);
+        this.state = {activeMenuItem : "(none)"};
+        this.clickItem = this.clickItem.bind(this);
+    }
+
+    clickItem(menuItem, e) {
+       // console.log("Am " + e.target.value);
+       this.setState({activeMenuItem : menuItem});
+    }
+
+    render() {
+        const rows = [];
+        if (typeof this.props.menuItems !== "undefined") {
```

Comment:
this if is too complicated from my perspective. here are some reasons:
1. comparing against the "undefined" string is considered a hack
2. doing this comparison `if (menuItem === this.state.activeMenuItem)` could get out of control because it compares the entire objects. what if the code evolves in a way that's not expected and the 2 object, although different should actually be the same?
3. code duplication when pushing the components inside `rows`

How I would propose to tackle these kind of scenarios (ps: the code is not tested):
```
const menuItems = this.props.menuItems || [];   // here you get free check for props.menuItems done by JS
const rows = menuItems.map(menuItem => {
  const className = menuItem.isActive ? 'selectedMenuItem': '';
  return (<div onClick={this.clickItem.bind(this, menuItem)} className={className}>{menuItem.name}</div>);
});
```